cmd/prepare-storage.go

```diff
@@ -49,18 +50,26 @@ var printEndpointError = func() func(Endpoint, error) {
 
 // Migrates backend format of local disks.
 func formatXLMigrateLocalEndpoints(endpoints EndpointList) error {
-	for _, endpoint := range endpoints {
+	g := errgroup.WithNErrs(len(endpoints))
+	for index, endpoint := range endpoints {
 		if !endpoint.IsLocal {
 			continue
 		}
-		formatPath := pathJoin(endpoint.Path, minioMetaBucket, formatConfigFile)
-		if _, err := os.Stat(formatPath); err != nil {
-			if os.IsNotExist(err) {
-				continue
+		index := index
+		g.Go(func() error {
+			epPath := endpoints[index].Path
+			formatPath := pathJoin(epPath, minioMetaBucket, formatConfigFile)
+			if _, err := os.Stat(formatPath); err != nil {
+				if os.IsNotExist(err) {
+					return nil
+				}
+				return err
 			}
-			return err
-		}
-		if err := formatXLMigrate(endpoint.Path); err != nil {
+			return formatXLMigrate(epPath)
+		}, index)
+	}
```

Comment:
It is risky to migrate the backend in parallel, because if it fails on one of the disks, we return and exit. Hence other migration go routines will be killed in the middle of migration causing the backend to be in undesirable state. Better do it serially, so that if something goes wrong, debugging is easier.

Reply:
Why is it risky? there is no middle of migration @krishnasrinivas - we either migrate or not, if you are saying we are not atomic then it will happen even serially too if some kills the server. 

Other go-routines don't die because of one error on one disk, they all complete on rest of the disks and we simply return an error if one of the disks failed. This way we don't stop in between i.e we finish all disks if possible. 

I think this approach is fine. This is also only for local endpoints and i don't see why it would be an issue to debug either.

Reply:
> Other go-routines don't die because of one error on one disk, they all complete on rest of the disks and we simply return an error if one of the disks failed.

I see, then it is fine.